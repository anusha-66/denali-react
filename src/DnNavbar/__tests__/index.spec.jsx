/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnNavbar, DnIcon } from '../../index'; // src/index.tsx
import brandImage from '../__stories__/denali-logo.svg';

describe('DnNavbar Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnNavbar />);

        expect(rendered).toMatchSnapshot();
    });

    it('should render default storybook example', () => {
        expect.assertions(1);

        const rendered = render(
            <DnNavbar>
                <DnNavbar.Left>
                    <DnNavbar.Brand src={brandImage} alt="Denali brand" />
                </DnNavbar.Left>
                <DnNavbar.Responsive>
                    <DnNavbar.Center>
                        <DnNavbar.Item isActive>Three</DnNavbar.Item>
                        <DnNavbar.Item>Center</DnNavbar.Item>
                        <DnNavbar.Item>Items</DnNavbar.Item>
                    </DnNavbar.Center>
                    <DnNavbar.Right>
                        <DnNavbar.Item href=".">
                            <DnIcon icon="dashboard" />
                        </DnNavbar.Item>
                        <DnNavbar.Item href=".">
                            <DnIcon icon="menu-dialpad" />
                        </DnNavbar.Item>
                        <DnNavbar.Item>
                            <DnIcon icon="user-profile-circle" />
                        </DnNavbar.Item>
                    </DnNavbar.Right>
                </DnNavbar.Responsive>
            </DnNavbar>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render brand without alt', () => {
        expect.assertions(1);

        const rendered = render(
            <DnNavbar>
                <DnNavbar.Left>
                    <DnNavbar.Brand src={brandImage} />
                </DnNavbar.Left>
            </DnNavbar>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
