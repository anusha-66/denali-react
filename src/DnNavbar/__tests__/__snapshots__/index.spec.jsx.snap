// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnNavbar Tests should render brand without alt 1`] = `
<nav
  class="nav"
>
  <div
    class="nav-left"
  >
    <img
      alt=""
      class="nav-brand"
      src="APP_ROOT/DnNavbar/__stories__/denali-logo.svg"
    />
  </div>
</nav>
`;

exports[`DnNavbar Tests should render default storybook example 1`] = `
<nav
  class="nav"
>
  <div
    class="nav-left"
  >
    <img
      alt="Denali brand"
      class="nav-brand"
      src="APP_ROOT/DnNavbar/__stories__/denali-logo.svg"
    />
  </div>
  <div
    class="nav-responsive"
  >
    <div
      class="nav-center"
    >
      <span
        class="nav-item is-active"
      >
        Three
      </span>
      <span
        class="nav-item"
      >
        Center
      </span>
      <span
        class="nav-item"
      >
        Items
      </span>
    </div>
    <div
      class="nav-right"
    >
      <a
        class="nav-item"
        href="."
      >
        <i
          class="d-icon d-dashboard is-default"
        />
      </a>
      <a
        class="nav-item"
        href="."
      >
        <i
          class="d-icon d-menu-dialpad is-default"
        />
      </a>
      <span
        class="nav-item"
      >
        <i
          class="d-icon d-user-profile-circle is-default"
        />
      </span>
    </div>
  </div>
</nav>
`;

exports[`DnNavbar Tests should render with default props 1`] = `
<nav
  class="nav"
/>
`;
